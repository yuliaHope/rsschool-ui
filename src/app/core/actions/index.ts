export * from './schedule';
export * from './user';
export * from './courses';
export * from './assignments';
