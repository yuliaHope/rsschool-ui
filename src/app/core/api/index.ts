export * from './schedule';
export * from './session';
export * from './user';
export * from './courses';
export * from './assignments';
