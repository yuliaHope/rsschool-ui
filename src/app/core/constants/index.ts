export * from './schedule';
export * from './courses';
export * from './user';
export * from './assignments';
