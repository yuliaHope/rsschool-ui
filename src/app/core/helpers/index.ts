export * from './schedule';
export * from './assignment';
