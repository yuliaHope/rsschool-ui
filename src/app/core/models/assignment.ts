export enum AssignmentStatus {
    Assigned = 'Assigned',
    ReadyForReview = 'ReadyForReview',
    Rejected = 'Rejected',
    Checked = 'Checked',
}

export interface IAssignment {
    taskId: number;
    courseId: string;
    studentId: string;
    mentorId: string;
    studentComment: string;
    mentorComment: string;
    score: number;
    assignmentRepo: string;
    deadlineDate: number;
    completeDate: number;
    checkDate: number;
    status: AssignmentStatus;
    [key: number]: any;
}

export interface IAssignmentDocument extends IAssignment {
    _id: string;
}
