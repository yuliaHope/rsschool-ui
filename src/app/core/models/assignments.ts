export enum AssignmentStatus {
    Assigned = 'Assigned',
    Checked = 'Checked',
    Rejected = 'Rejected',
    ReadyForReview = 'ReadyForReview',
    MissedDeadline = 'MissedDeadline',
}

export interface IAssignment {
    taskId: string;
    title: string;
    courseId: string;
    studentId: string;
    mentorId: string;
    studentComment: string;
    mentorComment: string;
    score: number;
    assignmentRepo: string;
    deadlineDate: number;
    completeDate: number;
    checkDate: number;
    status: AssignmentStatus;
}

export interface IAssignmentModel extends IAssignment {
    _id: string;
}
