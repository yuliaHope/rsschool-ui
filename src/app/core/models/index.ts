export * from './events';
export * from './stage';
export * from './user';
export * from './course';
export * from './assignments';
export * from './courseMentor';
export * from './courseStudent';
