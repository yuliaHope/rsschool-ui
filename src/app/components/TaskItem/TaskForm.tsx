import * as React from 'react';
import { IAssignmentModel } from 'core/models';

type Props = {
    status: string;
    assignment: IAssignmentModel;
    fetchAssignmentSolution: (assignment: IAssignmentModel) => void;
};

class TaskForm extends React.Component<Props> {
    private studentRepo: React.RefObject<HTMLInputElement>;
    private studentComment: React.RefObject<HTMLTextAreaElement>;
    constructor(props: Props) {
        super(props);
        this.studentRepo = React.createRef();
        this.studentComment = React.createRef();
    }

    SendStudentTask(assignment: IAssignmentModel) {
        const { fetchAssignmentSolution } = this.props;
        const studentRepoValue: string = this.studentRepo.current!.value.replace(/\s/g, '');
        if (studentRepoValue) {
            fetchAssignmentSolution(assignment);
        } else {
            alert('wrong repo');
            this.studentRepo.current!.value = '';
        }
    }

    render() {
        const { status } = this.props;
        if (status !== 'MissedDeadline' && status !== 'Checked' && status !== 'ReadyForReview') {
            const { assignment } = this.props;
            return (
                <small className="text-muted">
                    <form
                        onSubmit={ev => {
                            ev.preventDefault();
                            this.SendStudentTask(assignment);
                        }}
                    >
                        <div className="form-group">
                            <label>Choose repo</label>
                            <input
                                className="form-control form-control-sm"
                                type="text"
                                ref={this.studentRepo}
                                placeholder="Enter link"
                                pattern="https?://github.com/.+"
                            />
                        </div>
                        <div className="form-group">
                            <label>Comments</label>
                            <textarea
                                className="form-control form-control-sm"
                                ref={this.studentComment}
                                placeholder="Write comments here"
                            />
                        </div>
                        <button type="submit" className="btn btn-primary btn-sm">
                            Submit
                        </button>
                    </form>
                </small>
            );
        } else {
            return <React.Fragment />;
        }
    }
}

export default TaskForm;
