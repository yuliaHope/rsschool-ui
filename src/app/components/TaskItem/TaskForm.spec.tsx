import * as React from 'react';
import { shallow } from 'enzyme';
import TaskForm from './TaskForm';
import { AssignmentStatus } from '../../core/models';

describe('Task Item', () => {
    const props = {
        status: AssignmentStatus.Assigned,
        assignment: {
            _id: '_id',
            assignmentRepo: 'assignmentRepo',
            checkDate: 123,
            completeDate: 123,
            courseId: 'courseId',
            deadlineDate: 123,
            mentorComment: 'mentorComment',
            mentorId: 'mentorId',
            score: 0,
            status: AssignmentStatus.Assigned,
            studentComment: 'studentComment',
            studentId: 'studentId',
            taskId: 'taskId',
            title: 'title',
        },
        fetchAssignmentSolution: jest.fn(),
    };
    describe('render Task Form', () => {
        it('in case status is Assigned', () => {
            const output = shallow(<TaskForm {...props} />);
            expect(output).toMatchSnapshot();
        });
        it('in case status is Checked', () => {
            const nextProps = {
                ...props,
                status: AssignmentStatus.Checked,
                assignment: {
                    ...props.assignment,
                    status: AssignmentStatus.Checked,
                },
            };
            const output = shallow(<TaskForm {...nextProps} />);
            expect(output).toMatchSnapshot();
        });
    });
});
