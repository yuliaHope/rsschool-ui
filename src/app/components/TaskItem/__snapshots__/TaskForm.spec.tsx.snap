// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task Item render Task Form in case status is Assigned 1`] = `
<small
  className="text-muted"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label>
        Choose repo
      </label>
      <input
        className="form-control form-control-sm"
        pattern="https?://github.com/.+"
        placeholder="Enter link"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <label>
        Comments
      </label>
      <textarea
        className="form-control form-control-sm"
        placeholder="Write comments here"
      />
    </div>
    <button
      className="btn btn-primary btn-sm"
      type="submit"
    >
      Submit
    </button>
  </form>
</small>
`;

exports[`Task Item render Task Form in case status is Checked 1`] = `<React.Fragment />`;
